### PR TITLE
Option to automatically include all field names in fingerprint

### PR DESCRIPTION
**Pull request by hartfordfive**
Although I realize this can definitely be more computationally expensive, it does come in handy when you may have many types of events, that may change in structure on a regular basis.

### CODE REVIEW

File: lib/logstash/filters/fingerprint.rb [hunk 1]
```diff
@@ -154,7 +165,8 @@ def anonymize_openssl(data)
       hash  = OpenSSL::HMAC.digest(@digest, @key, data.to_s)
       Base64.strict_encode64(hash).force_encoding(Encoding::UTF_8)
     else
-      OpenSSL::HMAC.hexdigest(@digest, @key, data.to_s).force_encoding(Encoding::UTF_8)
+      hash = OpenSSL::HMAC.hexdigest(@digest, @key, data.to_s).force_encoding(Encoding::UTF_8)
```
Comment (jordansissel) [inline lib/logstash/filters/fingerprint.rb]: this is not necessary; Ruby has implicit returns  meaning the last expression's value is returned automatically.

**Comment by jordansissel** [comment]
Code and docs LGTM. Waiting on Travis CI.

**Comment by jordansissel** [comment]
One minor comment about a style change (change added an explicit return, was previously implicit)

File: lib/logstash/filters/fingerprint.rb (hunk 1 above)
Comment (hartfordfive) [inline lib/logstash/filters/fingerprint.rb]: Yes sorry, I only had that during my debugging.  Meant to take that out, will do that right now.

**Comment by hartfordfive** [comment]
@jordansissel Once this is accepted and merged, will this plugin version be compatible with Logstash 5.4.3?   As a potential feature, it would be nice if there was some type of page that indicates which plugin (at least the official Elastic ones) versions are compatible with which Logstash version.

**Comment by jordansissel** [comment]
It should be compatible, yes.

In terms of documenting the version matrix of (plugins x logstash), we are working on that, but I have no ETA.

**Pull request closed without merging**